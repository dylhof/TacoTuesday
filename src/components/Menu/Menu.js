import React from 'react';
import {Drawer, List, ListItem, ListSubheader, Typography} from '@material-ui/core';
import { Link } from 'react-router-dom'

const Menu = (props) => {
  console.log(props.showMenu)
  return (
    <Drawer open={props.showMenu} onClose={props.toggleShowMenu}>
      <div
      role="button"
      onClick={props.toggleShowMenu}
      onKeyDown={props.toggleShowMenu}>

          <Link to='/'><Typography variant='display2'>Taco Home</Typography></Link>
          <List>
            <ListSubheader>Explore</ListSubheader>
            <ListItem><Link to='/explore/baseLayers'>Base-Layers</Link></ListItem>
            <ListItem><Link to='/explore/mixins'>Mixins</Link></ListItem>
            <ListItem><Link to='/explore/condiments'>Condiments</Link></ListItem>
            <ListItem><Link to='/explore/seasonings'>Seasonings</Link></ListItem>
            <ListItem><Link to='/explore/shells'>Shells</Link></ListItem>
          </List>
          <Link to='/build_a_taco'><Typography variant='display2'>Build A Taco</Typography></Link>
      </div>
    </Drawer>
  )
}

export default Menu;