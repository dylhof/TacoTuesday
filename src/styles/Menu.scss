.menu {
  background-color: $lowlight;
  height: 100vh;
  
  h3 {
    @include text($title, 30px, left);
    margin: 15px;
  }
  li{
    @include text($copy, 20px, left);
    margin: 10px 10px 10px 30px;
    list-style-type: none;
  }
  a {
    color: $text;
    text-decoration: none;
    &:visited {
      color: $text;
    }
  }
}