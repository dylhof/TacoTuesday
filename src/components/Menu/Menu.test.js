import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu'

describe('Menu', () => {
  const mocktoggleShowMenu = jest.fn()

  const wrapper = shallow(
    <Menu 
    toggleShowMenu={mocktoggleShowMenu}
    showMenu={true}/>)

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
