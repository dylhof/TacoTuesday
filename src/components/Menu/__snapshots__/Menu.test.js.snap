// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<WithStyles(Drawer)
  className="drawer"
  onClose={[MockFunction]}
  open={true}
>
  <div
    className="menu"
    onClick={[MockFunction]}
    onKeyDown={[MockFunction]}
  >
    <Link
      replace={false}
      to="/"
    >
      <h3>
        I'll taco you Home
      </h3>
    </Link>
    <ul>
      <h3>
        Explore:
      </h3>
      <li>
        <Link
          replace={false}
          to="/explore/baseLayers"
        >
          Base-Layers
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/explore/mixins"
        >
          Mixins
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/explore/condiments"
        >
          Condiments
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/explore/seasonings"
        >
          Seasonings
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/explore/shells"
        >
          Shells
        </Link>
      </li>
    </ul>
    <Link
      replace={false}
      to="/build_a_taco"
    >
      <h3>
        Build A Taco
      </h3>
    </Link>
  </div>
</WithStyles(Drawer)>
`;
